lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -405,6 +405,7 @@ imsi64_t amf_app_handle_initial_ue_message(
 
         /* remove amf_ngap_ud_id entry from ue context */
         amf_remove_ue_context(ue_context_p);
+        old_amf_ue_ngap_id =  ue_context_p->amf_ue_ngap_id;
```

Comment:
I think rather than storing the value here like this and using on line 427, line 409 `ue_context_p->amf_ue_ngap_id = INVALID_AMF_UE_NGAP_ID;` should be removed. inside `amf_ue_context_update_coll_keys`, `ue_context_p->amf_ue_ngap_id` should be updated.

Reply:
Changes reverted as per comments and made changes as discussed over slack